### PR TITLE
Stop using variants storage for HTTP/REST/PropertyList sections

### PR DESCRIPTION
**Pull request by d-ronnqvist**
Bug/issue #, if applicable: 

## Summary

This changes the different `DocumentationNode` properties for HTTP/REST/PropertyList data to basic optional values instead of `DocumentationDataVariants`. 

As far as I know and can tell, these types of symbols never have other language representations so it's not necessary to store their information as language varying. Removing the possibility of language variability for these properties make them easier to work with on the documentation node level.

For the same reason, the `redirects` is also changed to a basic optional value instead of `DocumentationDataVariants`. It's not possible to author language-representation-specific redirects.

## Dependencies

None.

## Testing

Nothing in particular. This isn't a user-facing change.

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] ~Added~ Updated tests
- [x] Ran the `./bin/test` script and it succeeded
- ~[ ] Updated documentation if necessary~

### CODE REVIEW

**Comment by d-ronnqvist** [comment]
@swift-ci please test

File: Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorSymbolVariantsTests.swift [hunk 1]
```diff
@@ -500,72 +501,53 @@ class RenderNodeTranslatorSymbolVariantsTests: XCTestCase {
...
+        func propertiesSection(for dictionaryKeysSection: DictionaryKeysSection) throws -> PropertiesRenderSection {
+            symbol.dictionaryKeysSection = dictionaryKeysSection
+            context.documentationCache[dictionaryReference] = node
             
-            guard renderNode.primaryContentSections.indices.contains(propertiesSectionIndex) else {
-                XCTFail("Missing properties section")
-                return PropertiesRenderSection(title: "Properties", items: [])
-            }
+            let converter = DocumentationNodeConverter(bundle: bundle, context: context)
+            let renderNode = try converter.convert(node)
```
Comment (patshaughnessy) [inline Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorSymbolVariantsTests.swift:525]: This generates a new warning: `No calls to throwing functions occur within 'try' expression`.

Looks like you avoided the existing test code in `assertMultiVariantArticle` ? And rewrote this test directly inline?

**Review by patshaughnessy — APPROVED** [review]
Great simplification! Please just avoid the 2 new warnings.

File: Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorSymbolVariantsTests.swift (hunk 1 above)
Comment (d-ronnqvist) [reply Tests/SwiftDocCTests/Rendering/RenderNodeTranslatorSymbolVariantsTests.swift]: That warning wasn't there in https://github.com/swiftlang/swift-docc/pull/1104/commits/4709dec537375c9be44acdc473eebf5e196fd5c1 😞 . It got introduced after merging `main`. Still, it's fixed in https://github.com/swiftlang/swift-docc/pull/1104/commits/276e7db5b9a66ac8395b68f203c60c163597d3c8

**Comment by d-ronnqvist** [comment]
@swift-ci please test

**Pull request merged**